### PR TITLE
Change default value of branch parameters to main

### LINKED ISSUE #8045: Change default value for the branch parameter to 'main'
Tasks:

- [x] Add migration script to change branches without a value to 'master'. Old sources have branch parameters without value. Apparently, the default value of parameters has not always been inserted into reports. 
- [x] Also investigate whether the `Branch` and `Branches` parameter can be combined. In that case no default value would be needed.

**Combining the `Branch` and `Branches` parameters**

The `Branch` parameter is only used by `GitLab` and `AzureDevOps`, both for the metric: `source_up_to_dateness`.

The `Branches` parameters is only used by `GitLab` for the metric: `pipeline_duration`.

Note that `SonarQube` also has a branch parameter, but it's not derived from `Branch`, but rather a separate `StringParameter`. The analysis below only covers the use of `Branch` by `GitLab` and `AzureDevOps`.

To decide whether `Branch` and `Branches` can be combined it's important to look at the default values of the parameters. `Branches` has as default value "all branches". `Branch` has `master` (after this issue is completed: `main`) as default value. 

When configuring `GitLab` or `Azure DevOps` as source for the `source_up_to_dateness` metric, users can measure the up-to-dateness of files and of pipelines. The `Branch` parameter is only used in case of the files option: users can specify for which branch the up-to-dateness of the file should be measured. 

If we would replace the `Branch` parameter with the `Branches` parameter, *Quality-time* is basically saying that by default users will want to track the up-to-dateness of files across all branches. We haven't had that request from users.

Alternatively, we could change the default value of the `Branches` parameter to `main`. That would impact only the `pipeline_duration` metric when used with `GitLab` as source. Users that want to measure the duration of all pipelines regardless of branch would have to delete `main` from the parameter value. It seems more likely that users would want to measure the duration of all pipelines instead of just the main branch.

Getting rid of the `Branch` parameter would save five lines of code in the data model.

Conclusion: merging the two parameters would worsen user experience in return for a very small reduction of code size. It's not worth it.

### PR DESCRIPTION
**Pull request by fniessink**
Make the default value of branch parameters `main` instead of `master`.

Note: no migration code is needed because Quality-time only uses the default value of the branch parameter upon creation of a source. Changing the default value of the branch parameter does not impact existing sources.

Closes #8045.

### CODE REVIEW

File: components/collector/tests/source_collectors/azure_devops/test_source_up_to_dateness.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ async def test_age_of_file(self):
         self.assert_measurement(
             response,
             value=self.expected_age,
-            landing_url=f"{self.url}/_git/repo?path=README.md&version=GBmaster",
+            landing_url=f"{self.url}/_git/repo?path=README.md&version=GBmain",
```
Comment (wkoot) [inline components/collector/tests/source_collectors/azure_devops/test_source_up_to_dateness.py:40]: Does this come from the change in `shared_data_model.parameters.Branch`; is this the only instance of that class?

Comment (fniessink) [reply components/collector/tests/source_collectors/azure_devops/test_source_up_to_dateness.py:40]: Yes and yes :-)

Azure DevOps only uses the branch parameter for the source-up-to-dateness metric. There are other branch parameters used by Azure DevOps, namely branches to ignore and target branches to include, but they are lists.

Comment (wkoot) [reply components/collector/tests/source_collectors/azure_devops/test_source_up_to_dateness.py:40]: So, can the second part of #8045 be achieved?

Comment (fniessink) [reply components/collector/tests/source_collectors/azure_devops/test_source_up_to_dateness.py:40]: Oops. Forgot about that part. I'll have a look!

Comment (fniessink) [reply components/collector/tests/source_collectors/azure_devops/test_source_up_to_dateness.py:40]: I don't think it's worth it. See analysis in #8045. Agree?

Comment (wkoot) [reply components/collector/tests/source_collectors/azure_devops/test_source_up_to_dateness.py:40]: Yep makes sense! At best we could make `SonarQube` use `Branch`.. in a different story ;) Let's merge this?

**Pull request merged**